docs/extending.rst

```diff
@@ -183,6 +184,89 @@ periods (like a Python import path) and recursively uses the previous value to
 look up the next value until a final value is found.
 
 
+Subpreparers & Collections
+--------------------------
+
+Sometimes, your data isn't completely flat but is instead nested. This
+frequently occurs in conjunction with related data, such as a foreign key'd
+object or many-to-many scenario. In this case, you can lever "subpreparers".
+Restless ships with two of these, the ``SubPreparer`` & the
+``CollectionSubPreparer``.
+
+The ``SubPreparer`` is useful for a single nested relation. You define a
+regular ``Preparer/FieldsPreparer`` (perhaps in a shareable location), then
+use the ``SubPreparer`` to pull it in & incorporate the nested data. For
+example::
+
+    # We commonly expose author information in our API as nested data.
+    # This definition can happen in its own module or wherever needed.
+    author_preparer = FieldsPreparer(fields={
+        'id': 'pk',
+        'username': 'username',
+        'name': 'get_full_name',
+    })
+
+    # ...
+
+    # Then, in the main preparer, pull them in using `SubPreparer`.
+    preparer = FieldsPreparer(fields={
+        'author': SubPreparer('user', author_preparer),
+        # Other fields can come before/follow as normal.
+        'content': 'post',
+        'created': 'created_at',
+    })
+
+This results in output like::
+
+    {
+        "content": "Isn't my blog cool? I think so...",
+        "created": "2017-05-22T10:34:48",
+        "author": {
+            "id": 5,
+            "username": "joe",
+            "name": "Joe Bob"
+        }
+    }
+
+The ``CollectionSubPreparer`` operates on the same principle (define a set
+of fields to be nested), but works with collections of things. These collections
+should be ordered & behave similar to iterables like ``list``s & ``tuples``.
+As an example::
+
+    # Set up a preparer that handles the data for each thing in the broader
+    # collection.
+    # Again, this can be in its own module or just wherever it's needed.
+    comment_preparer = FieldsPreparer(fields={
+        'comment': 'comment_text',
+        'created': 'created',
+    })
+
+    # Use it with the ``CollectionSubPreparer`` to create a list
+    # of prepared sub items.
+    preparer = FieldsPreparer(fields={
+        # A normal blog post field.
+        'post': 'post_text',
+        # All the comments on the post.
+        'comments': CollectionSubPreparer('comments.all', comment_preparer),
```

Comment:
We could merge that an than work to make sure that #44 works fine. In that case the realize would be blocked by it. What do you think?